### DISCUSSION TITLE
在合约里，如何实现类似Map，Set，List类型效果？

### DISCUSSION THREAD

**Discussion by lispking** [openbuildxyz/mina-zkapp-bootcamp#222] [Q&A] [answered]
翻了文档半天，没找到合约的Map，Set，List类型，实验一下Field[]，发现直接报合约不认识，要实现类似的效果要怎么办？

比如要实现银行开户，限定一个人只能开一个户，那这时候就要用map来记录开户记录

**Comment by logere** [comment, marked as answer]
链上状态只能存8个32B的数据

**Reply by logere** [reply]
超过了就只能 off-chain-state https://docs.minaprotocol.com/zkapps/writing-a-zkapp/introduction-to-zkapps/how-zkapps-work#off-chain-state

**Reply by lispking** [reply]
这篇文章昨天翻到了，所以说mina上只能走离线的方式来实现类似的效果
